load_all_pokemon.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+
+import collections
+import os
+from database import Pokemon
+
+SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+DATA_DIR = os.path.join(SCRIPT_DIR, 'Data')
+IMAGES_DIR = os.path.join(SCRIPT_DIR, 'Images')
+EXTRA_DIR = os.path.join(IMAGES_DIR, 'Extra')
+
+region_info = collections.namedtuple('region_info', 'start end dir_name')
+region_dict = {
+    'kanto': region_info(1, 151, 'Generation I - Kanto'),
+    'johto': region_info(152, 251, 'Generation II - Johto'),
+    'hoenn': region_info(252, 386, 'Generation III - Hoenn'),
+    'sinnoh': region_info(387, 493, 'Generation IV - Sinnoh'),
+    'extra': region_info(0, 0, 'Extra')
+}
+
+g_pokemon_dict = None  # potential fathers for extra pokemon
+
+
+def region_name_by_id(id):
+    if not id:
+        return 'extra'
+    for name, region in region_dict.items():
+        if region.start <= int(id) <= region.end:
+            return name
+    assert False, 'region_name_by_id({})'.format(id)
+
+
+def make_a_pokemon(i, line):
+    id = '{:03}'.format(i + 1)
+    line = line.strip().split()
+    while len(line) < 4:  # add '' as the subtype if it is missing
+        line.append('')
+    name, threshold, main_type, subtype = line
+    name = name.lower()
+    threshold = float(threshold)
+    region = region_name_by_id(id)
+    dir_name = region_dict[region].dir_name
+    path = os.path.join(IMAGES_DIR, dir_name, id + '.jpg')
+    return Pokemon(id, name, region, path, main_type, subtype, threshold)
+
+
+def load_pokemon(filename='pokemon.txt'):
+    """Load everything but the Pokemon from the 'Extra' folder"""
+    with open(os.path.join(DATA_DIR, filename)) as in_file:
+        return [make_a_pokemon(i, line) for i, line in enumerate(in_file)]
```

Comment:
Wow!  Yes you can!,    `print(list(enumerate('Apple', 1)))` I never knew that one.